src/apps/loan-list/list/loan-list.tsx

```diff
@@ -70,67 +108,134 @@ const LoanList: FC<LoanListProps> = ({ pageSize }) => {
     }
   }, [publizonData]);
 
+  const openLoanDetailsModal = useCallback(
+    (modalId: string) => {
+      setModalDetailsId(modalId);
+      open(`${loanDetails}${modalId}`);
+    },
+    [loanDetails, open]
+  );
+
+  const openDueDateModal = useCallback(
+    (dueDateInput: string) => {
+      setDueDate(dueDateInput);
+      open(`${dueDateModal}${dueDateInput}`);
+    },
+    [dueDateModal, open]
+  );
+
   useEffect(() => {
     const modalString = getUrlQueryParam("modal");
+    // if there is a loan details query param, loan details modal should be opened
+    if (modalString && modalString.includes(loanDetails as string)) {
+      const loanDetailsModalId = getLoanDetailsModalId(modalString);
+      if (loanDetailsModalId) {
+        const id = idFromLoanDetailsModalString(loanDetailsModalId);
+        if (id) {
+          setModalDetailsId(id);
+        }
+      }
+    }
 
     // modal query param: modal loans all
     if (modalString === allLoansId) {
       open(allLoansId);
     }
-  }, [physicalLoans, open, allLoansId]);
+
+    // If there is a query param with the due date, a modal should be opened
+    if (modalString && containsDueDateModalString(modalString)) {
+      const dateFromQueryParam = dateFromDueDateModalString(modalString);
+      if (dateFromQueryParam) {
+        openDueDateModal(dateFromQueryParam);
+      }
+    }
+  }, [allLoansId, loanDetails, open, openDueDateModal]);
 
   return (
-    <div className="loan-list-page">
-      <h1 className="text-header-h1 my-32">{t("loanListTitleText")}</h1>
-      {((Array.isArray(physicalLoans) && physicalLoans.length > 0) ||
-        (Array.isArray(digitalLoans) && digitalLoans.length > 0)) && (
-        <>
-          {physicalLoans && (
-            <List
-              pageSize={pageSize}
-              emptyListLabel={t("loanListPhysicalLoansEmptyListText")}
-              loans={physicalLoans}
-              dueDates={physicalLoansDueDates}
-              view={view}
-            >
-              <ListHeader
-                header={t("loanListPhysicalLoansTitleText")}
-                amount={physicalLoans.length}
+    <>
+      <div
+        style={modalIds.length > 0 ? { display: "none" } : {}}
+        className="loan-list-page"
+      >
+        <h1 className="text-header-h1 my-32">{t("loanListTitleText")}</h1>
+        {((Array.isArray(physicalLoans) && physicalLoans.length > 0) ||
+          (Array.isArray(digitalLoans) && digitalLoans.length > 0)) && (
```

Comment:
For the readability you could consider moving this away from the jsx and put it before the return with a variablename called something like: `const listContainsLoans = ...`

Reply:
done